[#Wave-Control: Add NetSuite]: Top-level Subsidiary selection

mananjadhav opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

Covers the implementation of [Top-level subsidiary config page](https://docs.google.com/document/d/1WubNv_VAv78IxG4FKsi9aS0pWESfWvUYbqBAAy5b4bc/edit#heading=h.12ncvrmo0icg).

This is one of the first FE PRs for NetSuite without the proper auth flow integrated. Hence, to get this working we'll need to have a few pre-requisites:

1. Collect workspace with accounting enabled.
2. `NetSuite` beta enabled for the user.  You can return true in the [canUseNetSuiteIntegration](https://github.com/Expensify/App/blob/6294e7d3462abe4124e6486de6702727b78a2c2c/src/libs/Permissions.ts#L51-L53) method.
3. Setup the netsuite data linked in the comment [here](https://github.com/Expensify/App/issues/43435#issuecomment-2159929615) for the given workspace as `connections: {netsuite: <linkedjson>}`. 
> One way I did the setup is calling `Onyx.merge` with `useEffect` on any FE component.
> ```
>    useEffect(() => {
>        Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}76A10AB364F95DA0`, {"connections":{"netsuite":{"accountID":"TSTDRV1668486", ....})
>    }, []);
>
> ``` 


### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
5. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/43435
PROPOSAL:


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
6. Verify a modal appears displaying a preview of that image
--->


1. Open the app with the pre-setup workspace.
2. Go to `Accounting` from the LHS
3. Verify that you're able to see `Net Suite` in the active connections along with the `Oracle NetSuite` logo.
4. Verify that you're able to see the `Subsidiary` list item below NetSuite title along with the subsidiary listed as  `Honeycomb Mfg.` (based on the test data imported)
5. Click on the item, and see that it opens the list of subsidiaries in RHS. 
6. Confirm the title and the description of the screen.
7. Click on one of the subsidiaries.
8. It should take you back to the Accounting screen, with subsidiary listed based on the selection in step 7.

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->


1. Open the app with the pre-setup workspace.
2. Go to `Accounting` from the LHS
3. Go offline.
4. Verify that you're able to see `Net Suite` in the active connections along with the `Oracle NetSuite` logo.
5. Verify that you're able to see the `Subsidiary` list item below NetSuite title along with the subsidiary listed as  `Honeycomb Mfg.` (based on the test data imported)
6. Click on the item, and see that it opens the list of subsidiaries in RHS. 
7. Confirm the title and the description of the screen.
8. Click on one of the subsidiaries.
9. It should take you back to the Accounting screen, with subsidiary listed based on the selection in step 8 but in grey disabled color.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
11. Upload an image via copy paste
12. Verify a modal appears displaying a preview of that image
--->

Same as Test Steps. Additionally do a sanity on list pages such as `Report`, `Search` that the list and scroll behavior is not impacted.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3069065/ad39eca8-242a-4c3b-8032-7c050728ce04

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3069065/ad25cf64-4e4b-4d14-920e-612a18b5f8c2

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3069065/8d01e600-c922-4180-a998-737873813d9b

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/3069065/795e2512-8104-4ac4-8c87-91134583f7a6

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

**Basic Test**

https://github.com/Expensify/App/assets/3069065/d9e539e5-ba39-4397-8cb9-7062caacdc7b


**Offline Behavior**

https://github.com/Expensify/App/assets/3069065/bde3e1fe-e3cb-480c-b56c-208a261b4987


**Web Spanish Translation**

<img width="1511" alt="web-net-suite-subsidiary-es" src="https://github.com/Expensify/App/assets/3069065/44a56e11-2f61-4a3f-9c98-8bb6bca2306a">


**Error handling**

https://github.com/Expensify/App/assets/3069065/1759ccb2-f42a-4368-987a-c53984e360da

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/3069065/cbd6ae31-40ae-403f-b1da-79bf3f64e69d

</details>

mananjadhav committed:
Revert "Merge pull request #42965 from suneox/40767-handle-edit-last-message"

This reverts commit 478db524560497b58da5b430740a5396f1add1f1, reversing
changes made to bb14c990ca1bbb654d68ebeb2ea95dd8afcb9b5c.

mananjadhav committed:
Reapply "Merge pull request #42965 from suneox/40767-handle-edit-last-message"

This reverts commit 747fe1941f019ee95a474deb2c0f808eedbcd9ac.

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App

mananjadhav committed:
feat: net suite setup

mananjadhav committed:
feat: navigation types

mananjadhav committed:
feat: page setup with options

mananjadhav committed:
fix: default selection

mananjadhav committed:
fix: routing

mananjadhav committed:
feat: show selected subsidiary

mananjadhav committed:
style: lint fixes

mananjadhav committed:
refactor: remove console.log

mananjadhav committed:
feat: added write command

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
refactor: rename type

mananjadhav committed:
refactor: lint fixes

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
feat: added icon and es content

mananjadhav committed:
fix: added fallback

mananjadhav committed:
refactor: rename net_suite identifier

mananjadhav committed:
feat: added api types

mananjadhav committed:
feat: modify optimistic data

mananjadhav committed:
feat: added api call

mananjadhav committed:
fix: param name

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav commented:
@shubham1206agra The PR is WIP. I'll ask for the review once it's ready.

mananjadhav commented on src/libs/actions/connections/netsuite.ts:
```diff
@@ -0,0 +1,91 @@
+                        netsuite: {
+                            options: {
+                                config: {
+                                    subsidiary,
```
WIP.

mananjadhav replied:
Added. Thanks for noticing this.

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav commented on src/libs/API/parameters/UpdateNetSuiteSubsidiaryParams.ts:
```diff
@@ -0,0 +1,6 @@
+type UpdateNetSuiteSubsidiaryParams = {
+    policyID: string;
+    subsidiary: string;
```
I don't think it's necessary to pass `authToken`.

mananjadhav committed:
refactor: update id handling

mananjadhav committed:
fix: added id in params

mananjadhav committed:
fix: added comment

mananjadhav committed:
fix: added pending action

mananjadhav committed:
fix: added error handling

mananjadhav committed:
style: lint fixes

mananjadhav committed:
feat: nested scrollview handling

mananjadhav committed:
style: lint fixes

mananjadhav commented on src/libs/actions/connections/netsuite.ts:
```diff
@@ -0,0 +1,91 @@
+                        netsuite: {
+                            options: {
+                                config: {
+                                    subsidiary: oldSubsidiary,
```
Added. Thanks for noticing this.

mananjadhav commented on src/libs/actions/connections/netsuite.ts:
```diff
@@ -0,0 +1,91 @@
+                        netsuite: {
+                            options: {
+                                config: {
+                                    subsidiary,
```
Will keep this, and replace once your PR is merged for the common method.

mananjadhav committed:
refactor: error handling

mananjadhav commented:
@shubham1206agra @yuwenmemon The PR is now ready for review.

@Expensify/design Would love to get your feedback. Especially on the error scenario. I am not sure where should the error message should up. In the `Accounting` page or `Subsidiary`. In Subsidiary too if the user has selected an option at the bottom of the page, they won't see the error at the top.

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
style: lint fixes

mananjadhav commented on src/pages/workspace/accounting/netsuite/NetSuiteSubsidiarySelector.tsx:
```diff
@@ -0,0 +1,95 @@
+            shouldIncludeSafeAreaPaddingBottom
+            connectionName={CONST.POLICY.CONNECTIONS.NAME.NETSUITE}
+        >
+            <OfflineWithFeedback
+                errors={ErrorUtils.getLatestErrorField(netsuiteConfig ?? {}, CONST.NETSUITE_CONFIG.SUBSIDIARY)}
+                errorRowStyles={[styles.ph5, styles.mt2]}
+                onClose={() => Policy.clearNetSuiteErrorField(policyID, CONST.POLICY.CONNECTIONS.NAME.NETSUITE, CONST.NETSUITE_CONFIG.SUBSIDIARY)}
+            >
```
I've asked design input on this one. In Xero we're showing the error for the organization name screen.

mananjadhav replied:
I think the current design is approved so I think I'll resolve this comment now.

mananjadhav commented on src/pages/workspace/accounting/netsuite/NetSuiteSubsidiarySelector.tsx:
```diff
@@ -0,0 +1,95 @@
+    };
+
+    return (
+        <ConnectionLayout
```
To use `OfflineWithFeedback`.

mananjadhav commented on src/pages/workspace/accounting/netsuite/NetSuiteSubsidiarySelector.tsx:
```diff
@@ -0,0 +1,95 @@
+
+NetSuiteSubsidiarySelector.displayName = 'NetSuiteSubsidiarySelector';
+
+export default withPolicy(NetSuiteSubsidiarySelector);
```
Will update.

mananjadhav replied:
Updated.

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
refactor: use withpolicyconnections

mananjadhav commented on src/pages/workspace/accounting/netsuite/NetSuiteSubsidiarySelector.tsx:
```diff
@@ -0,0 +1,95 @@
+    const currentSubsidiaryID = netsuiteConfig?.subsidiaryID ?? '';
+    const policyID = policy?.id ?? '';
+
+    const sections =
+        subsidiaryList.map((subsidiary: NetSuiteSubsidiary) => ({
+            text: subsidiary.name,
+            keyForList: subsidiary.name,
+            isSelected: subsidiary.name === currentSubsidiaryName,
+            subsidiaryID: subsidiary.internalID,
+        })) ?? [];
```
I don't think this will be needed other places. So don't see a point to add in utils. If I see reusing this, then I'll move it to utils then

mananjadhav commented on src/libs/actions/Policy/Policy.ts:
```diff
@@ -526,6 +526,10 @@ function clearXeroErrorField(policyID: string, fieldName: string) {
     Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, {connections: {xero: {config: {errorFields: {[fieldName]: null}}}}});
 }
 
+function clearNetSuiteErrorField(policyID: string, connectionName: ConnectionName, fieldName: string) {
+    Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, {connections: {[connectionName]: {options: {config: {errorFields: {[fieldName]: null}}}}}});
+}
```
Yes agreed.  Will update.

mananjadhav replied:
Updated.

mananjadhav commented on src/libs/actions/connections/netsuite.ts:
```diff
@@ -0,0 +1,99 @@
+                        netsuite: {
+                            options: {
+                                config: {
+                                    subsidiary: newSubsidiary.subsidiary,
+                                    subsidiaryID: newSubsidiary.subsidiaryID,
```
Yeah I anyway need to update this once your PR is in, will update at that time.

dannymcclain commented:
>@Expensify/design Would love to get your feedback. Especially on the error scenario. I am not sure where should the error message should up. In the Accounting page or Subsidiary. In Subsidiary too if the user has selected an option at the bottom of the page, they won't see the error at the top.

What's shown in your video seems pretty right to me... @shawnborton @dubielzyk-expensify any other ideas how we'd treat this? I don't really know what to do about the case where there's a long list and the user has one selected down towards the bottom.

shawnborton commented:
Hmm yeah, I agree with Danny - I think what you have works? Any additional thoughts @trjExpensify?

How exactly is this kind of error triggered in the first place?

mananjadhav commented:
> How exactly is this kind of error triggered in the first place?

API error.

trjExpensify commented:
What's the API error for, when it comes to the subsidiary selection?

mananjadhav commented:
> What's the API error for, when it comes to the subsidiary selection?

Wouldn't be 100% sure, a safe guard. Could be any API failure.

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
refactor: update args for error field

mananjadhav commented:
@shubham1206agra @yuwenmemon This is ready for the next pass of the review. Can we close this on priority? It is blocking other PR work.

mananjadhav committed:
Merge branch 'main' of github.com:mananjadhav/App into mj-netsuite-subsidiary-select

mananjadhav committed:
refactor: rename imports and files

mananjadhav committed:
fix: go back

mananjadhav committed:
refactor: run prettier

mananjadhav commented on src/pages/workspace/accounting/netsuite/NetSuiteSubsidiarySelector.tsx:
```diff
@@ -0,0 +1,95 @@
+                subsidiaryID: currentSubsidiaryID,
+            },
+        );
+        Navigation.goBack(ROUTES.WORKSPACE_ACCOUNTING.getRoute(policyID));
```
Okay updated. The fix works.

mananjadhav commented on src/libs/actions/connections/netsuite.ts:
```diff
@@ -0,0 +1,99 @@
+import Onyx from 'react-native-onyx';
```
Done.

mananjadhav commented:
The console error seems unrelated to changes in this one. I'll first fix the SelectionScreen as I am at the airport. If time permits I'll try to fix the console.error.

mananjadhav committed:
refactor: use selectionscreen

mananjadhav committed:
refactor: api call fixes

mananjadhav committed:
refactor: import lint error

mananjadhav committed:
refactor: rename methods and error conditions

mananjadhav committed:
fix: Spanish translation

mananjadhav committed:
refactor: added fallback

mananjadhav committed:
fix: rollback unwanted prop

mananjadhav committed:
fix: remove extra space

mananjadhav committed:
fix: netsuite name

mananjadhav committed:
refactor: remove unwanted types

mananjadhav commented on src/pages/workspace/accounting/netsuite/NetSuiteSubsidiarySelector.tsx:
```diff
@@ -0,0 +1,95 @@
+                onClose={() => Policy.clearNetSuiteErrorField(policyID, CONST.NETSUITE_CONFIG.SUBSIDIARY)}
+            >
+                <Text style={[styles.ph5, styles.pb5]}>{translate('workspace.netsuite.subsidiarySelectDescription')}</Text>
+            </OfflineWithFeedback>
```
Yes updated with `SelectionScreen`.  Also removed the extra prop I added for nesting. Thanks.

mananjadhav commented on src/pages/workspace/accounting/netsuite/NetSuiteSubsidiarySelector.tsx:
```diff
@@ -0,0 +1,110 @@
+import type SCREENS from '@src/SCREENS';
+import type {NetSuiteSubsidiary} from '@src/types/onyx/Policy';
+
+type NetSuiteSubsidiarySelectorProps = WithPolicyConnectionsProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.ACCOUNTING.NETSUITE_SUBSIDIARY_SELECTOR>;
```
Removed.

mananjadhav committed:
fix: lint errors

mananjadhav commented:
Revised screencasts with `SelectionScreen`.

**Web** Also contains empty screen screenshot.


https://github.com/Expensify/App/assets/3069065/04812fc7-00f3-48ba-8d83-0f49f48bc899



**iOS**: Doesn't have the console.error so I removed the SelectionList prop that I added for nested view.

https://github.com/Expensify/App/assets/3069065/2d7dcf1c-53a0-4bac-b34a-ff17eff4af40

yuwenmemon reviewed (approved):
Nice!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9783702010) to production by https://github.com/jasperhuangg in version: 9.0.3-7 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9872741083) to production by https://github.com/Julesssss in version: 9.0.5-13 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

ikevin127 commented on src/pages/workspace/accounting/PolicyAccountingPage.tsx:
```diff
@@ -234,6 +254,27 @@ function PolicyAccountingPage({policy, connectionSyncProgress}: PolicyAccounting
                       },
                   ]
                 : []),
+            ...(policyConnectedToNetSuite && !shouldShowSynchronizationError
+                ? [
+                      {
+                          description: translate('workspace.netsuite.subsidiary'),
+                          iconRight: Expensicons.ArrowRight,
+                          title: netSuiteSelectedSubsidiary,
+                          wrapperStyle: [styles.sectionMenuItemTopDescription],
+                          titleStyle: styles.fontWeightNormal,
+                          shouldShowRightIcon: netSuiteSubsidiaryList.length > 1,
+                          shouldShowDescriptionOnTop: true,
+                          pendingAction: policy?.connections?.netsuite?.options?.config?.pendingFields?.subsidiary,
+                          brickRoadIndicator: policy?.connections?.netsuite?.options?.config?.errorFields?.subsidiary ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined,
+                          onPress: () => {
+                              if (!(netSuiteSubsidiaryList.length > 1)) {
+                                  return;
+                              }
+                              Navigation.navigate(ROUTES.POLICY_ACCOUNTING_NETSUITE_SUBSIDIARY_SELECTOR.getRoute(policyID));
+                          },
+                      },
+                  ]
+                : []),
```
Looks like this part of the logic led to the following issue:
- https://github.com/Expensify/App/issues/48274#issuecomment-2368480825

Which we solved by returning empty object when there's no value for subsidiary, more details in issue [proposal](https://github.com/Expensify/App/issues/48274#issuecomment-2318192111).
